src/vs/workbench/browser/parts/editor/editorsObserver.ts

```diff
@@ -359,9 +360,17 @@ export class EditorsObserver extends Disposable {
 
 			return true;
 		});
+		let editorsToCloseCount: number;
+		if (ignoreDirtyTab === true) {
+			editorsToCloseCount = leastRecentlyClosableEditors.length - limit + 1;
```

Comment:
Why the +1?
```suggestion
			editorsToCloseCount = leastRecentlyClosableEditors.length - limit;
```

Reply:
I set  Workbench.editor.limit to 5 and enable ignoreDirtyTab,   and I have 3 dirty files ,then I can only open 4 files when without "+1";  and add "+1", I can open  5 files.   The reson should be that leastRecentlyClosableEditors don't include the sticky tab